src/coreclr/vm/gchelpers.cpp

```diff
@@ -513,9 +513,11 @@ OBJECTREF TryAllocateFrozenSzArray(MethodTable* pArrayMT, INT32 cElements)
 
     // The initial validation is copied from AllocateSzArray impl
 
-    if (pArrayMT->ContainsPointers() && cElements > 0)
+    if (pArrayMT->Collectible() || (pArrayMT->ContainsPointers() && cElements > 0))
```

Comment:
Maybe make it a separate if so that it's easier to read?

Reply:
There are several places in this file using similar condition checks, so I would think it is not necessary.